### PR TITLE
Sbromberger/performance

### PR DESCRIPTION
**Pull request by sbromberger**
Some modest performance improvements via changes to vector construction. Benchmarks below.

### CODE REVIEW

**Comment by sbromberger** [comment]
Current master:
```
julia> @benchmark maximal_cliques(g)
BenchmarkTools.Trial:
  memory estimate:  25.01 MiB
  allocs estimate:  262716
  --------------
  minimum time:     82.230 ms (3.74% GC)
  median time:      98.771 ms (4.19% GC)
  mean time:        105.826 ms (4.23% GC)
  maximum time:     273.944 ms (2.80% GC)
  --------------
  samples:          48
  evals/sample:     1

julia> @benchmark Graph{UInt128}(2_000_000)
BenchmarkTools.Trial:
  memory estimate:  167.85 MiB
  allocs estimate:  2000003
  --------------
  minimum time:     88.527 ms (0.00% GC)
  median time:      300.513 ms (51.51% GC)
  mean time:        369.063 ms (71.75% GC)
  maximum time:     818.394 ms (83.02% GC)
  --------------
  samples:          15
  evals/sample:     1

julia> @benchmark DiGraph{UInt128}(2_000_000)
BenchmarkTools.Trial:
  memory estimate:  339.18 MiB
  allocs estimate:  4000043
  --------------
  minimum time:     200.655 ms (0.00% GC)
  median time:      234.941 ms (0.00% GC)
  mean time:        500.348 ms (54.32% GC)
  maximum time:     2.084 s (87.78% GC)
  --------------
  samples:          14
  evals/sample:     1


julia> g = loadsnap(:as_caida)
{26475, 53381} undirected simple Int64 graph

julia> @benchmark bellman_ford_shortest_paths(g,1:4000)
BenchmarkTools.Trial:
  memory estimate:  1.59 MiB
  allocs estimate:  146
  --------------
  minimum time:     5.067 ms (0.00% GC)
  median time:      7.900 ms (0.00% GC)
  mean time:        9.431 ms (2.27% GC)
  maximum time:     95.981 ms (0.00% GC)
  --------------
  samples:          530
  evals/sample:     1
```

**Comment by sbromberger** [comment]
This PR:
```
julia> @benchmark maximal_cliques(g)
BenchmarkTools.Trial:
  memory estimate:  24.06 MiB
  allocs estimate:  236151
  --------------
  minimum time:     71.611 ms (3.82% GC)
  median time:      76.603 ms (3.80% GC)
  mean time:        82.643 ms (4.14% GC)
  maximum time:     191.367 ms (4.12% GC)
  --------------
  samples:          61

julia> @benchmark Graph{UInt128}(2_000_000)
BenchmarkTools.Trial:
  memory estimate:  167.85 MiB
  allocs estimate:  2000003
  --------------
  minimum time:     72.036 ms (0.00% GC)
  median time:      288.544 ms (72.73% GC)
  mean time:        354.420 ms (76.01% GC)
  maximum time:     730.508 ms (85.93% GC)
  --------------
  samples:          16
  evals/sample:     1

julia> @benchmark DiGraph{UInt128}(2_000_000)
BenchmarkTools.Trial:
  memory estimate:  335.69 MiB
  allocs estimate:  4000005
  --------------
  minimum time:     144.234 ms (0.00% GC)
  median time:      166.987 ms (0.00% GC)
  mean time:        405.505 ms (61.52% GC)
  maximum time:     1.873 s (91.58% GC)
  --------------
  samples:          14
  evals/sample:     1

julia> g = loadsnap(:as_caida)
{26475, 53381} undirected simple Int64 graph

julia> @benchmark bellman_ford_shortest_paths(g,1:4000)
BenchmarkTools.Trial:
  memory estimate:  1.58 MiB
  allocs estimate:  136
  --------------
  minimum time:     5.327 ms (0.00% GC)
  median time:      7.015 ms (0.00% GC)
  mean time:        7.729 ms (2.43% GC)
  maximum time:     15.877 ms (22.21% GC)
  --------------
  samples:          646
  evals/sample:     1
```

**Comment by sbromberger** [comment]
I will be merging this this week, unless there are objections

cc @jpfairbanks @juliohm

File: src/graphtypes/simplegraphs/simpledigraph.jl [hunk 1]
```diff
@@ -16,12 +16,8 @@ eltype(x::SimpleDiGraph{T}) where T = T
 
 # DiGraph{UInt8}(6), DiGraph{Int16}(7), DiGraph{Int8}()
 function (::Type{SimpleDiGraph{T}})(n::Integer = 0) where T<:Integer
-    fadjlist = Vector{Vector{T}}()
-    badjlist = Vector{Vector{T}}()
-    for _ = one(T):n
-        push!(badjlist, Vector{T}())
-        push!(fadjlist, Vector{T}())
-    end
+    fadjlist = [Vector{T}() for _ = one(T):n]
```
Comment (juliohm) [inline src/graphtypes/simplegraphs/simpledigraph.jl]: Just curious, will this `one(T)` go away in Julia v0.7 and beyond?

Comment (sbromberger) [reply src/graphtypes/simplegraphs/simpledigraph.jl]: I don't know. Was there talk of deprecating it?

**Comment by sbromberger** [comment]
Thanks, @juliohm!

File: src/graphtypes/simplegraphs/simpledigraph.jl (hunk 1 above)
Comment (juliohm) [reply src/graphtypes/simplegraphs/simpledigraph.jl]: I don't' know, I was just curious why we need `one(T)` in this range expression when it doesn't participate in the `Vector{T}()` part. Should the argument to the function be `n::T` instead or you are allowing different integer types on purpose here for type coercion?

Comment (sbromberger) [reply src/graphtypes/simplegraphs/simpledigraph.jl]: We don't necessarily need `one(T)` here. I loosened `n` precisely for type coercion.

**Pull request merged**